ext/oss/sink.go

```diff
@@ -200,6 +225,35 @@ func (o *OSSSink) process() error {
 		return errors.WithStack(err)
 	}
 
+	if o.enableArchive {
+		err := o.DryRunable(func() error {
+			pathsToArchive := []string{}
+			for destinationURI := range o.writeHandlers {
+				tmpPath, err := getTmpPath(destinationURI)
+				if err != nil {
+					o.Logger().Error(fmt.Sprintf("failed to get tmp path for %s: %s", destinationURI, err.Error()))
+					return errors.WithStack(err)
+				}
+
+				pathsToArchive = append(pathsToArchive, tmpPath)
+			}
+			o.Logger().Info(fmt.Sprintf("compressing %d files: %s", len(pathsToArchive), strings.Join(pathsToArchive, ", ")))
+
+			archivePaths, err := o.archive(pathsToArchive)
```

Comment:
archive function should doing archive only, can we do the upload the archived files outside this function?